src/HookTrait.php

```diff
@@ -173,24 +175,20 @@ public function hook(string $spot, array $args = [])
 
             try {
                 while ($_data = array_pop($this->hooks[$spot])) {
-                    foreach ($_data as $index => &$data) {
-                        $return[$index] = call_user_func_array(
-                            $data[0],
-                            array_merge(
-                                [$this],
-                                $args,
-                                $data[1]
-                            )
-                        );
+                    foreach ($_data as $index => $data) {
+                        $return[$index] = $data[0](...array_merge(
```

Comment:
Isn't it possible that $data[0] is defined in array format like `[$this, 'methodName']` ?

Reply:
https://3v4l.org/E5cZ5

Reply:
Oh perfect then!
I see my old-school php 5.x habits here :D